src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -497,9 +497,11 @@ public void visitToken(DetailAST ast) {
     private boolean isVerifiable(DetailAST methodOrVariableDef) {
         boolean result = true;
         if (!memberModifiers.isEmpty()) {
-            final DetailAST modifiersAst = methodOrVariableDef
-                    .findFirstToken(TokenTypes.MODIFIERS);
-            result = isContainVerifiableType(modifiersAst);
+            if (methodOrVariableDef.getType() != TokenTypes.RECORD_COMPONENT_DEF) {
```

Comment:
`methodOrVariableDef.getType() != TokenTypes.RECORD_COMPONENT_DEF`
now variable name `methodOrVariableDef` looks weird, please rename it to something abstract like `token` or `ast`

Please also correct javadoc